sw/device/lib/testing/rand_testutils.c

```diff
@@ -82,5 +83,24 @@ uint32_t rand_testutils_gen32_range(uint32_t min, uint32_t max) {
   if (range == 0) {
     return min;
   }
-  return min + (rand_testutils_gen32() % (range + 1));
+  uint32_t result = min + (rand_testutils_gen32() % (range + 1));
+  CHECK(result >= min && result <= max);
+  return result;
+}
+
+void rand_testutils_shuffle(void *array, size_t size, size_t length) {
+  if (length <= 1) {
+    return;
+  }
+  unsigned char temp[size];
+  unsigned char *array8 = array;
+  uint32_t reseed_frequency = rand_testutils_rng_ctx.reseed_frequency;
+  rand_testutils_rng_ctx.reseed_frequency = UINT32_MAX;
+  for (size_t i = length - 1; i > 0; i--) {
+    size_t j = rand_testutils_gen32_range(0, i + 1);
+    memcpy(temp, array8 + j * size, size);
+    memcpy(array8 + j * size, array8 + i * size, size);
+    memcpy(array8 + i * size, temp, size);
```

Comment:
Is it overkill to use `memcpy` to swap a single byte at one index for a byte at a different index? Can't you just swap the bytes without a `memcpy`?

Reply:
oh sorry I misread, I thought you were just swapping one byte, not multiple (which you are)